src/spec-common/injectHeadless.ts

```diff
@@ -528,24 +538,14 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 					return runSingleCommand(command, name);
 				});
 			}
+
 			await Promise.all(commands);
-		} catch (err) {
+		} catch {
```

Comment:
I think the error needs to be rethrown after reporting progress to preserve the previous behavior. Only if it is rethrown will the "further user-provided commands" be skipped. Please check.